src/App.tsx

```diff
@@ -1,27 +1,62 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const loadAllGoods = (): void => {
+    getAll()
+      .then(goodsFromServer => {
+        setGoods(goodsFromServer);
+      });
```

Comment:
1. Show loader when fetching
2. Catch and show error on error
3. Prevent clicking on the same button twice (multiple requests for the same data)
4. Prevent clicking all buttons when loading

Fix it all